pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,358 @@
+package ftp
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/textproto"
+	"os"
+	"path"
+	"slices"
+	"time"
+
+	"gofr.dev/pkg/gofr/container"
+
+	"github.com/jlaffaye/ftp"
+)
+
+// Conn struct embeds the *ftp.ServerConn returned by ftp server on successful connection.
+type Conn struct {
+	*ftp.ServerConn
+}
+
+// Retr wraps the ftp retrieve method to return a ftpResponse interface type.
+func (c *Conn) Retr(filepath string) (ftpResponse, error) {
+	return c.ServerConn.Retr(filepath)
+}
+
+func (c *Conn) RetrFrom(filepath string, offset uint64) (ftpResponse, error) {
+	return c.ServerConn.RetrFrom(filepath, offset)
+}
+
+var (
+	errEmptyFilename      = errors.New("filename cannot be empty")
+	errEmptyPath          = errors.New("file/directory path cannot be empty")
+	errEmptyDirectory     = errors.New("directory name cannot be empty")
+	errInvalidArg         = errors.New("invalid filename/directory")
+	transferCompleteError = &textproto.Error{Code: 226, Msg: "Transfer complete."}
```

Comment:
```suggestion
	transferCompleteError = &textproto.Error{Code: textproto.StatusClosingDataConnection, Msg: "Transfer complete."}
```

But would simply remove this line

```suggestion
```

Because I'm suggesting this https://github.com/gofr-dev/gofr/pull/871#discussion_r1700719709